components/camel-huaweicloud-smn/src/main/java/org/apache/camel/component/huaweicloud/smn/SimpleNotificationProducer.java

```diff
@@ -0,0 +1,366 @@
+        PublishMessageResponse response;
+
+        PublishMessageRequestBody apiBody;
+        this.clientConfigurations = validateServiceConfigurations(endpoint, exchange);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Checking operation name");
+        }
+        switch (clientConfigurations.getOperation()) {
+
+            case SmnOperations.PUBLISH_AS_TEXT_MESSAGE:
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Publishing as text message");
+                }
+                apiBody = new PublishMessageRequestBody()
+                        .withMessage(exchange.getMessage().getBody(String.class))
+                        .withSubject(clientConfigurations.getSubject())
+                        .withTimeToLive(String.valueOf(clientConfigurations.getMessageTtl()));
+
+                response = smnClient.publishMessage(new PublishMessageRequest()
+                        .withBody(apiBody)
+                        .withTopicUrn(clientConfigurations.getTopicUrn()));
+                break;
+
+            case SmnOperations.PUBLISH_AS_TEMPLATED_MESSAGE:
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Publishing as templated message");
+                }
+                apiBody = new PublishMessageRequestBody()
+                        .withMessage(exchange.getMessage().getBody(String.class))
+                        .withSubject(clientConfigurations.getSubject())
+                        .withTimeToLive(String.valueOf(clientConfigurations.getMessageTtl()))
+                        .withMessageTemplateName((String) exchange.getProperty("CamelHwCloudSmnTemplateName"))
+                        .withTags((HashMap<String, String>) exchange.getProperty("CamelHwCloudSmnTemplateTags"))
+                        .withTimeToLive(String.valueOf(clientConfigurations.getMessageTtl()));
+
+                response = smnClient.publishMessage(new PublishMessageRequest()
+                        .withBody(apiBody)
+                        .withTopicUrn(clientConfigurations.getTopicUrn()));
+                break;
+
+            default:
+                throw new UnsupportedOperationException(
+                        String.format("operation %s not supported in publishMessage service",
+                                clientConfigurations.getOperation()));
+        }
+        setResponseParameters(exchange, response);
+    }
+
+    /**
+     * maps api response parameters as exchange property
+     * 
+     * @param exchange
+     * @param response
+     */
+    private void setResponseParameters(Exchange exchange, PublishMessageResponse response) {
+        if (response == null) {
+            return; // mapping is not required if response object is null
+        }
+        if (!ObjectHelper.isEmpty(response.getMessageId())) {
+            exchange.setProperty("CamelSmnMesssageId", response.getMessageId());
```

Comment:
I would prefer to separate the headers properties and exchange properties into a separate constant class, e.g: `SimpleNotificationConstants`. You can take a look at other components for reference

Reply:
Done. Moved all constants to SmnProperties class